pkg/workceptor/kubernetes_test.go

```diff
@@ -1,75 +1,95 @@
 package workceptor
 
 import (
-	"context"
+	"os"
+	"reflect"
 	"testing"
-
-	"github.com/ansible/receptor/pkg/netceptor"
+	"time"
 )
 
-func Test_isCompatibleK8S(t *testing.T) {
-	type args struct {
-		version      string
-		isCompatible bool
-	}
+func TestShouldUseReconnect(t *testing.T) {
+	const envVariable string = "RECEPTOR_KUBE_SUPPORT_RECONNECT"
 
-	kw := &kubeUnit{
-		BaseWorkUnitForWorkUnit: &BaseWorkUnit{},
+	tests := []struct {
+		name     string
+		envValue string
+		want     bool
+	}{
+		{
+			name:     "Positive (undefined) test",
+			envValue: "",
+			want:     false,
+		},
+		{
+			name:     "Enabled test",
+			envValue: "enabled",
+			want:     true,
+		},
+		{
+			name:     "Disabled test",
+			envValue: "disabled",
+			want:     false,
+		},
+		{
+			name:     "Auto test",
+			envValue: "auto",
+			want:     false,
+		},
+		{
+			name:     "Default test",
+			envValue: "default",
+			want:     false,
+		},
 	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.envValue != "" {
```

Comment:
If your interested, this if/else block can be removed by turning `envValue` into a function pointer, and in each test case you `os.Setenv` if needed

Reply:
Here is an example of it being done this way: https://github.com/ansible/receptor/blob/devel/pkg/workceptor/workceptor_test.go#L140